vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -100,67 +116,47 @@ def _set_targets(self):
             if not self.targets:
                 raise ValueError(f"Selected column {self.column} not found in any dataframe on this page.")
 
-    def _convert_column_type(self, data_frame):
-        if is_string_dtype(data_frame[self.column]):
-            if data_frame[self.column].str.contains(r"^\d{4}-\d{2}-\d{2}", regex=True).all():
-                data_frame[self.column] = pd.to_datetime(data_frame[self.column], format="mixed")
-
-        return data_frame
-
     def _set_column_type(self):
         data_frame = data_manager._get_component_data(self.targets[0])
-        data_frame = self._convert_column_type(data_frame=data_frame)
 
-        if is_datetime64_any_dtype(data_frame[self.column]):
-            self._column_type = "temporal"
-        elif is_numeric_dtype(data_frame[self.column]):
+        if is_numeric_dtype(data_frame[self.column]):
             self._column_type = "numerical"
+        elif is_datetime64_any_dtype(data_frame[self.column]):
+            self._column_type = "temporal"
         else:
             self._column_type = "categorical"
 
     def _set_selector(self):
         self.selector = self.selector or SELECTORS[self._column_type][0]()
         self.selector.title = self.selector.title or self.column.title()
 
-    def _set_slider_values(self):
-        if isinstance(self.selector, SELECTORS["numerical"]):
-            if self._column_type != "numerical":
-                raise ValueError(
-                    f"Chosen selector {self.selector.type} is not compatible "
-                    f"with {self._column_type} column '{self.column}'."
-                )
+    def _validate_allowed_selector(self):
+        if not isinstance(self.selector, ALLOWED_SELECTORS[self._column_type]):
+            raise ValueError(
+                f"Chosen selector {self.selector.type} is not compatible "
+                f"with {self._column_type} column '{self.column}'. "
+                f"Allowed selectors for {self._column_type} column '{self.column}' "
+                f"are {ALLOWED_SELECTORS[self._column_type]}."
```

Comment:
This is a bit too repetitive and verbose currently.
```suggestion
                f"Allowed selectors are {ALLOWED_SELECTORS[self._column_type]}."
```